pkg/test/framework/api/component/configuration.go

```diff
@@ -17,42 +17,42 @@ package component
 import "fmt"
 
 var (
-	_ Requirement = &ConfiguredRequirement{}
+	_ Requirement = &RequirementWrapper{}
 )
 
 // Configuration is a marker interface for configuration objects that components take.
 type Configuration fmt.Stringer
 
-// A configured requirement contains the requirement, a configuration object, and the name to give
-// that configured object.
-type ConfiguredRequirement struct {
-	name   string
-	req    Requirement
-	config Configuration
+// RequirementWrapper wraps a requirement and adds a name and a configuration.
+type RequirementWrapper struct {
```

Comment:
Looking at the uses of this, I wonder if it might be better if this were a builder.  Usage might be something like this:

reqA := NewRequirementBuilder().Name("bob").Component(component.PilotID).Configuration("...").Build()

Reply:
I played with a couple different versions, I think I like the functional version since it is shorter code. I don't expect this to grow beyond just name and config.

We could shorten even more:
reqA := Configure(Named(ids.Pilot, "bob"), myConfig)

Reply:
I could also add a separate "WrapRequirement" method that takes both name and config:

reqA := WrapRequirement(&ids.Pilot, "bob", myconfig)

That would simplify a bit of the code as well.  I could go back to that and not have separate Name vs. Configure methods. What do you think?

Builder still seems like overkill here :)

Reply:
yeah the functional version works.

Reply:
I am not sure I fully grok why you would need this (for config purposes)?

I'd expect the usage model to simply be:
```
c := component.RequireNamed(&ids.Echo, "b")
c.Configure(echo.Config{...})
```

To hold the name, we can simply make Requirements named, with default being empty string.

Reply:
It seems like there are two concerns here - configuration for a component of type X with id Y and the ability to create variations of components with distinct ids for equivalent types.

For the former I would expect to see something like

components.ConfigureGalley(context, &ids.Galley, <some strongly typed struct>)

For the latter I would expect to see

id = context.CreateAlias(&ids.Galley, "some name")

In order to customize initialization and dependency management you would see something like

galleyVariantId = context.CreateAlias(&ids.Galley, "galley-variant")
components.ConfigureGalley(context, galleyVariantId, { initFromPixieDust : true})
components.ConfigurePilot(context, &ids.Pilot, { galleyId : galleyVariantId })

Note that the parameter for pilot declares its dependency on the galley instance id which the component would then use to initialize itself appropriately.

Reply:
Chatted with Louis, let me take a pass at a more general model that also allows taking dependencies on dynamic variants, will ping the PR when updated with that.